### DISCUSSION TITLE
Dewheater continuosly or looped

### DISCUSSION THREAD

**Discussion by ruspa1978** [AllskyTeam/allsky-modules#89] [General]
Do you recommend using the dewheater continuously or looped once it activates by setting?

**Comment by troublestarter** [comment]
Hello @ruspa1978 
I'm not actually familiar with this setup (specially this module from allsky, i have developped my own in python), but can you please add some detail of your actual setup (wich hardware do you use and you current configuration for the dewheater module) ?

Thanks you.

**Reply by troublestarter** [reply]
For my side i have place the dewheater near the camera (the best place to be efficient)
![image](https://github.com/AllskyTeam/allsky/assets/6012203/b82e6274-79ef-445c-8dfd-f3692796c693)

I use heating resistors to have the best heat performance. Classic resistors are not really suitable to do that.
So when you calculate the dewpoint, you must ensure that you have 3 or 4 degrees minimum more than the dewpoint (dewpoint for inside and outside !) because the dew can be created inside the dome or outside the dome.

But i will not explain all here. You can have more details here and translate the page if needed: https://astrowick.fr/index.php/bricolages/developpement-informatique/20-developpement/86-gestion-conditionnement-allsky

I'm waiting for your details of your setup and configuration for the dewheater module

**Reply by ruspa1978** [reply]
Wow this is a complete solution.

**Comment by troublestarter** [comment]
I have a look on the program source code, 
What have you set on this parameter : `limit ` ?

It permit to : "If the temperature is within this many degrees of the dew point the heater will be enabled or disabled"

So if you set 6 (for LIMIT) and the actual temperature is only 2 degree more than dewpoint, you should let it heat **until** your temperature is more than 6 degree above the dewpoint.

**Reply by ruspa1978** [reply]
I set 6 for limit dewpoint

**Comment by Alex-developer** [comment]
I would not have the heater on continuously as there is little point. Set the limit and let the module do the work for yoy

**Comment by ruspa1978** [comment]
I use the round dewheather and the values of RH and Temp from cloudwather, so external. I don’t know if use it continuously, when it is activated with default settings, can damage something. I was thinking of having the sensor value read every 5 minutes and starting the dewheater for 3 minutes, intermittently.

**Reply by troublestarter** [reply]
Maybe @Alex-developer can add some detail on the dewheater module.
Could be cool.

But @ruspa1978 maybe you could see how long the dewheater is down and and how long is UP and how goes the temparature.

**Reply by Alex-developer** [reply]
Continuous is fine, mine run continuously and I have never had an issue

**Reply by ruspa1978** [reply]
This: https://www.dewcontrol.com/product/dew-heater-module---all-sky-camera
Black color

**Reply by ruspa1978** [reply]
I use continuously. Thanks

**Reply by dens50** [reply]
I have modified Alex's dewheater to use both openweather data for outside data and a SHT31 for inside data.  It runs closed-loop to keep the inside temperature at a fixed rise above outside dewpoint. Plus, a fixed rise above outside temp.  Heater does cycle on/off through the night.
But, I do agree with Alex, that the heater can be left on continuously - at night.

**Reply by troublestarter** [reply]
Oh great ! Can you share the code for @Alex-developer maybe ?
Should be interesting, dewheater for external and internal is important

**Reply by Alex-developer** [reply]
Dens50  - Sounds cool.  Happy to take a look at it if you would like it added to the core module

**Reply by dens50** [reply]
I am reluctant to post code. I hack ugly.  I arbitrarily deleted lines that didn't apply to me. 
I took 1.0.5 code and after getSensorReading (which is configed as openweather), read SHT31 for internal data. 
in the elif force check line, use internal temp.
in the following else, check internal temp is limit above external dewpoint
then mods to the logging and setting new variables = None

**Reply by Alex-developer** [reply]
No problem :-) 

Make sure you keep a backup copy of your code as any updates to the dewheater module may overwrite them.

**Reply by dens50** [reply]
Alex,
as a follow-up, off-topic question. will an update to allsky touch /opt/allsky? that is where the module code was put.
and/or are the modules updated separately from allsky?
Dennis

**Reply by Alex-developer** [reply]
Dens50

The modules in /opt/allsky/modules will ONLY be updated if you reinstall the extra modules. The main Allsky updater will not touch them.

This may change in the future but there are no immediate plans at the moment for it to change.
